x-pack/plugins/lists/server/services/items/delete_list_item.ts

```diff
@@ -36,6 +38,19 @@ export const deleteListItem = async ({
       },
       refresh,
     });
+
+    if (response.deleted) {
+      const checkIfListItemDeleted = async (): Promise<void> => {
+        const deletedListItem = await getListItem({ esClient, id, listItemIndex });
+        if (deletedListItem !== null) {
+          throw Error('List item has not been re-indexed in time');
+        }
+      };
+
+      await waitUntilDocumentIndexed(checkIfListItemDeleted);
+    } else {
+      throw Error('No list item has been deleted');
```

Comment:
Maybe include the id and index in these messages? A little more context in the message might also help:

```suggestion
      throw Error('Deletion of List Item [item_id] from [item_index] was not successful');
```